src/documentation_server.ml

```diff
@@ -0,0 +1,51 @@
+open Import
+
+type t = Polka.Server.t
+
+(** [debouncing_middleware ~timeout_in_ms f] on every request schedules [f] to
+    be run in [timeout_in_ms] milliseconds; on every new request the last
+    scheduled timeout is overridden *)
+let debouncing_middleware ~timeout_in_ms f =
+  let timeout = ref None in
+  fun ~request:_ ~response:_ ~next ->
+    (* on every request, we clear previous timeout and set a new one *)
+    Option.iter !timeout ~f:Node.clearTimeout;
+    timeout := Some (Node.setTimeout f timeout_in_ms);
```

Comment:
Since this function relies on staging, could you stage it explicitly with `Stage.t`?